### ISSUE TITLE
Issue with propagating trace from jms listener.

### ISSUE THREAD

**Issue by dennyjoe** [opentracing-contrib/java-jms#6]
Hello,

I have a JMS consumer that is annotated with `@JMSListener`. I successfully receive messages and opentracing shows the span in the trace. 

The issue is that this trace is not extended to anything that the consumer calls afterwards. Is this a defect, expected behaviour, or am I doing something wrong? 

Please review the trace below.

![jaegar-burndown](https://user-images.githubusercontent.com/28047547/31563881-ce158a5e-b058-11e7-8e79-778873f257e9.PNG)

If this is expected behaviour is it possible to join this one to the span related to the message listener. I can start a new span, but I don't know to join the two spans.

Here's my code for the application:

```java
@Configuration
@ComponentScan
@EnableAutoConfiguration
@SpringBootApplication
@EnableJms
public class Application extends SpringBootServletInitializer {

	public static void main(String[] args) {
		SpringApplication.run(Application.class, args);
	}
		
	@Bean
	public MessageConverter tracingJmsMessageConverter(Tracer tracer) {
	   return new TracingMessageConverter(new SimpleMessageConverter(), tracer);
	}

	@Bean
	public io.opentracing.Tracer jaegerTracer() {
		return new com.uber.jaeger.Configuration("project-burndown",
				com.uber.jaeger.Configuration.SamplerConfiguration.fromEnv(),
				com.uber.jaeger.Configuration.ReporterConfiguration.fromEnv()).getTracer();
	}
	
	@Bean
	public RestTemplate restTemplate(RestTemplateBuilder restTemplateBuilder) {
		return restTemplateBuilder.build();
	}

	@Bean
	public JmsTemplate jmsTemplate(ConnectionFactory connectionFactory, Tracer tracer) {
	   JmsTemplate jmsTemplate = new TracingJmsTemplate(connectionFactory, tracer);
	   jmsTemplate.setMessageConverter(new SimpleMessageConverter());
	   return jmsTemplate;
	}
	
	@Bean
	public JmsListenerContainerFactory<?> myFactory(ConnectionFactory connectionFactory,
			DefaultJmsListenerContainerFactoryConfigurer configurer) {
		DefaultJmsListenerContainerFactory factory = new DefaultJmsListenerContainerFactory();
		configurer.configure(factory, connectionFactory);
		return factory;
	}
}
```

This is my code for the listener. I've added in the spans, but I hope they are not necessary.

```java
@Component
public class NewProjectConsumer {

	@Autowired
	private Tracer tracer;
	
	@Autowired
	private ProjectBurndownService projectBurndownService;

	@JmsListener(destination = "new.project.topic", containerFactory = "myFactory")
	public void onMessage(String message) {
		ActiveSpan span = tracer.buildSpan("newProjectBurnDown").startActive();
		try {
			Project project = Project.fromJSON(message);
			span.setTag("project.id", project.getId());
			span.log("Creating new project burndown");
			projectBurndownService.newProject(project);
		} finally {
			span.deactivate();
		}
	}
}
```

Any advice is welcome.

Dennis

**Comment by malafeev** [comment]
@dennyjoe I'm working on that (there is a solution in [spring-cloud opentracing](https://github.com/opentracing-contrib/java-spring-cloud/pull/59), thinking about porting it here)

**Comment by malafeev** [comment]
fixed in 0.0.4
